Adds in the LookUpPort function call that returns a struct containing

ldemouy opened the pull request:
information about a port including it's description and keyword.
SSH and the web port are current placeholders for the recommended ports list

ldemouy committed:
Adds in the LookUpPort function call that returns a struct containing
information about a port including it's description and keyword.
SSH and the web port are current placeholders for the recommended ports list

ldemouy committed:
update library.pro to build the portlookup class
Refactor portlookup from a function that reads a file each time it is
called to a class that reads the file into a QStringList the first time
that it is called and reuses it afterwards.

ldemouy committed:
remove unnecessary <string> include, delete the QFile once we're done
with it. Add in a constructor that initializes the port cache, and add
in a destructor to delete the cache if portlookup is deleted.

beanpole135 commented:
Ok, one other thing I see which should probably get fixed before I can merge this in:
Port definitions are always  in pairs (number/type), but you currently ignore the type of port for both input/output. While it is common that both port types (udp/tcp) are registered to the same service, that is not necessarily true all the time, and firewall management still treats each port/type pair as the unique value.

For example:
port 2019 tcp is associated with the "whosockami" service, while port 2019 udp is associated with the "about" service (not necessarily the same functionality). If you look at other ports registered in the 2000-2050 range of the service file you will find lots of other example of different services using different types of the same port number.

ldemouy committed:
Make portlookup take the port type into consideration when checking the
port

ldemouy committed:
add the port type to the PortInfo struct

ldemouy committed:
Move PortInfo struct out of PortLookUp class, and fiddled with the regex
to make it actually work based on testing. The filter operation now will
pickup things like 522/tcp in the QStringList if you're searching for
22/tcp, but testing seems to indicate that just taking the first one in
the list works for getting what we want.

beanpole135 commented:
Ok, I think it looks fine now.
I do have a couple admin changes for it (listed below). Do you want to make these change before I merge it in, or do you mind if I make those changes after the merge?
1) Rename "PortLookup.h" to "sysadm-firewall.h" for consistency in the installed include files.
2) Convert the "PortLookup" class name to a more generic "Firewall" name in preparation for adding the rest of the firewall management functionality into it.

ldemouy committed:
renamed PortLookUp to firewall, and the files to sysadm-firewall

beanpole135 commented:
Great! Thanks for working on that!
